### PR TITLE
[dnf] Collect legacy yum config and dnf global vars

### PR DESCRIPTION
**Pull request by pmoravec**
- collect legacy yum->dnf symlinks
- collect dnf global vars
- obfuscate all passwords in all vars

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [X] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

File: sos/report/plugins/dnf.py [hunk 1]
```diff
@@ -55,7 +55,13 @@ def setup(self):
             '/etc/dnf/modules.d/.*.module': 'dnf_modules'
         })
 
-        self.add_copy_spec("/etc/dnf/")
+        self.add_copy_spec([
+            "/etc/dnf/",
+            "/etc/dnf/vars/",
```
Comment (TurboTurtle) [inline sos/report/plugins/dnf.py:60]: This should already be collected by the previous copyspec - unless we're hitting size limiting somewhere?

File: sos/report/plugins/dnf.py [hunk 2]
```diff
@@ -120,8 +126,9 @@ def setup(self):
         self.get_modules_info(modules['output'])
 
     def postproc(self):
-        regexp = r"(proxy_password(\s)*=(\s)*)(\S+)\n"
+        regexp = r"(password(\s)*=(\s)*)(\S+)\n"
```
Comment (TurboTurtle) [inline sos/report/plugins/dnf.py:129]: Does this need a leading `(.*)?` optional match in order to still hit `proxy_password` along with all others?

Comment (pmoravec) [reply sos/report/plugins/dnf.py:129]: Until we have there `^` or something similar, it should work well - as I checked:

    # cat /etc/dnf/vars/test.conf 
    proxy_password = test
    password = different
    # python3 bin/sos report -o dnf --batch --build
    ..
    # cat /var/tmp/sosreport-pmoravec-rhel8-2022-09-21-vfswezo/etc/dnf/vars/test.conf 
    proxy_password = ********
    password = ********
    #

File: sos/report/plugins/dnf.py (hunk 1 above)
Comment (pmoravec) [reply sos/report/plugins/dnf.py:60]: Ah, I was blind.. thanks for pointing to the dup.

**Review by TurboTurtle — APPROVED** [review]

**Pull request merged**